src/main/java/com/nike/cerberus/service/AuthenticationService.java

```diff
@@ -467,4 +485,23 @@ protected String getKeyId(IamPrincipalCredentials credentials) {
 
         return metadata;
     }
+
+    protected Optional<AwsIamRoleRecord> findIamRoleAssociatedWithSdb(final String iamPrincipalArn) {
+        Optional<AwsIamRoleRecord> iamRole = awsIamRoleDao.getIamRole(iamPrincipalArn);
+
+        try {
```

Comment:
Not sure we should have this try/catch or the one you added above.  What is the purpose?

As far as I can tell, if there is a bug in the code it will let things still partially function, but it would be better for the bug to be obvious so we can fix it, instead of obscure and hard to track down.

Though, I might have misunderstood the intention here.  Thanks.